### ISSUE TITLE
Trained Models

### ISSUE THREAD

**Issue by jasjeetIM** [yuqili3/NPGD_linear_inverse_prob#3]
Hello, 

Would it be possible to upload the MNIST and CelebA models that were used int the paper? I am trying to reproduce some experimental results. 

Thanks

**Comment by yuqili3** [comment]
Hi there Here is the link to the trained models on google drive: https://drive.google.com/drive/folders/1HaWhJUobaH3Z4zr7LiFBM26MRL-Frb8e?usp=sharing How to use:Download this folder. There are 2 subfolders called “MNIST_DCGAN_k100” and “CELEBA_DCGAN_k100”. In both subfolders you can find a generator model and a inverse generator model. The following code can be used to load the models: (take CELEBA as an example)```root_path = ‘your_path/trained_models'k=100 # this k refers to the embedding dimension kgen_path = '%s/CELEBA_DCGAN_k%d/generator_param.pkl'%(root_path, k)G = models.DCGAN_generator(k=k, d=128)G.load_state_dict(torch.load(gen_path))G.cuda() invG_path = '%s/CELEBA_DCGAN_k%d/invGen_param_mtl_loss_noisyGANoutput.pkl'%(root_path,k)invG = models.inverse_DCGAN(k=k,d=128)invG.load_state_dict(torch.load(invG_path))invG.cuda() G.eval()invG.eval()``` Hope this piece would help you. Please let me know if there is more questions.   Best regards,Yuqi Li From: jasjeetIMSent: Monday, October 12, 2020 12:21To: yuqili3/NPGD_linear_inverse_probCc: SubscribedSubject: [yuqili3/NPGD_linear_inverse_prob] Trained Models (#3) Hello,Would it be possible to upload the MNIST and CelebA models that were used int the paper? I am trying to reproduce some experimental results.Thanks—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by jasjeetIM** [comment]
Thank you for your quick response!

Two more questions:

1. Data:  

Assuming that each image in the celebA dataset has values in [0,255], how does the following  transform the range of values to [-1,1]? 
 https://github.com/yuqili3/NPGD_linear_inverse_prob/blob/6c3b033d8e117293e7c645146850c087b54af04c/CELEBA_DCGAN.py#L102-L105

Using definition of transforms from https://pytorch.org/docs/stable/torchvision/transforms.html. 

2. How many iterations Algorithm 1. did you run to get results in Fig. 7? In reference to: https://arxiv.org/pdf/1902.09698.pdf

Thanks in advance!

**Comment by yuqili3** [comment]
Hi there I cannot remember exactly which step does the pixel value is transformed from [0,255] to [0,1], maybe in this line of code. https://github.com/yuqili3/NPGD_linear_inverse_prob/blob/6c3b033d8e117293e7c645146850c087b54af04c/CELEBA_DCGAN.py#L110 or maybe we have already preprocessed the image data. It is best if you could check the pixel values in cuda. (I remember the pixel values is indeed in [-1,1] after the transform)40 Best regards,Yuqi Li From: jasjeetIMSent: Monday, October 12, 2020 14:38To: yuqili3/NPGD_linear_inverse_probCc: yuqi li; CommentSubject: Re: [yuqili3/NPGD_linear_inverse_prob] Trained Models (#3) Thank you for your quick response!Two more questions:Data:Assuming that each image in the celebA dataset has values in [0,255], how does the following transform the range of values to [-1,1]?https://github.com/yuqili3/NPGD_linear_inverse_prob/blob/6c3b033d8e117293e7c645146850c087b54af04c/CELEBA_DCGAN.py#L102-L105Using definition of transforms from https://pytorch.org/docs/stable/torchvision/transforms.html.How many iterations Algorithm 1. did you run to get results in Fig. 7? In reference to: https://arxiv.org/pdf/1902.09698.pdfThanks in advance!—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by jasjeetIM** [comment]
Thank you! Yes, pixel values are in [-1,1] in cuda. 

Can you also address the second question: 

How many iterations Algorithm 1. did you run to get results in Fig. 7? In reference to: https://arxiv.org/pdf/1902.09698.pdf

I am using your models, your code, and your celebA test data files and getting poor results after 500 iterations (attached)
![182639](https://user-images.githubusercontent.com/7819680/95786570-b796cc00-0c8c-11eb-9c7c-233e7b0afa4c.jpg)
![182638](https://user-images.githubusercontent.com/7819680/95786573-b82f6280-0c8c-11eb-86fe-46df4d8071ef.jpg)

Thanks

**Comment by yuqili3** [comment]
Yeah. I answered your second question in the last email: 40 epochs. And the step size (called lr in code) is ranging from 2 to 10.

Best regards
Yuqi Li

________________________________

**Comment by jasjeetIM** [comment]
I see, thank you. The format of your responses is showing up in a weird way: https://github.com/yuqili3/NPGD_linear_inverse_prob/issues/3

I will run Algorithm 1 for 40 iterations for each image and try different step_sizes (lr) in [0.5,10]. Is there a method you used to determine the step_size for each image? 

Context: I am working on a similar problem (generative models in CS) and would like to cite your paper if I am able to reproduce the results. Hence, any help is appreciated.

**Comment by yuqili3** [comment]
No not really. We used trial and error....

**Comment by jasjeetIM** [comment]
Understood. Would you happen to have step_sizes used for each image in celebAtest.zip? If not, then I will perform some sort of grid search and use the best results.

**Comment by yuqili3** [comment]
Sorry I could not find the exact step sizes for all images.

**Comment by jasjeetIM** [comment]
Hello, 

After significant effort, I am still not able to reproduce results of your paper - using your model, your test data, and your reconstruction hyper-parameters. Further, the reproduced results are not remotely close to the results in the paper. This is very surprising. Could you confirm that you the models you provided for CelebA are the ones used in your paper? Shouldn't one be able to reproduce results (atleast almost exactly) using your model, test data, and code? 

Edit: In case you are able to reproduce the results and I am not, I may be doing something wrong. Hence any advise here would be super helpful. 

Thanks

**Comment by FlorianPesce** [comment]
@jasjeetIM I can confirm, I have also been unable to reproduce results for CelebA.
